### PR TITLE
RQ-446: improve app routing

### PR DESCRIPTION
**Pull request by rohanmathur91**
Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/views/containers/MockServerContainer/MockServerSidebar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import PATHS from "config/constants/sub/paths";
+import "./MockServerSidebar.css";
+
+const mockServerSubRoutes = [
+  {
+    title: "My Mock Servers",
+    path: PATHS.MOCK_SERVER_V2.ABSOLUTE,
+  },
+  {
+    title: "My File Servers",
```
Comment (lazyvab) [inline app/src/views/containers/MockServerContainer/MockServerSidebar/index.tsx:12]: `My Files`

File: app/src/routes/index.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import { accountRoutes } from "./accountRoutes";
+import { authRoutes } from "./authRoutes";
+import { desktopRoutes } from "./desktopRoutes";
+import { mobileDebuggerRoutes } from "./mobileDebuggerRoutes";
+import { mockServerRoutes } from "./mockServerRoutes";
+import { onboardingRoutes } from "./onboardingRoutes";
+import { miscRoutes } from "./miscRoutes";
+import { settingRoutes } from "./settingRoutes";
+
+export const routes: RouteObject[] = [
```
Comment (lazyvab) [inline app/src/routes/index.ts:14]: +1 for the split.

File: app/src/views/containers/RulesContainer/RulesSidebar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <div className="rules-sidebar-container secondary-sidebar-container">
+      <ul>
+        {ruleRoutes.map(({ path, title }) => (
+          <li key={title}>
+            <NavLink
+              to={path}
+              className={({ isActive }) =>
+                isActive || (path === PATHS.RULES.MY_RULES.ABSOLUTE && pathname.includes(PATHS.RULE_EDITOR.RELATIVE))
```
[the remark this answers is not in the document]
Comment (rohanmathur91) [reply app/src/views/containers/RulesContainer/RulesSidebar/index.tsx:36]: style PR fixes this

File: app/src/utils/PathUtils.js [hunk 4]
```diff
@@ -2,6 +2,10 @@ import APP_CONSTANTS from "../config/constants";
 
 const { PATHS } = APP_CONSTANTS;
 
+export const joinPaths = (path1 = "", path2 = "") => {
+  return path1.concat(path2?.[0] !== "/" ? "/" + path2 : path2).replace(/\/\//g, "/");
```
[the remark this answers is not in the document]
Comment (rohanmathur91) [reply app/src/utils/PathUtils.js:6]: Its needed, consider a case `path1 = "/"` and `path2 = "/something"` without replace the path will be `"//something"`

**Pull request merged**